tests/foreman/api/test_registration.py

```diff
@@ -126,3 +127,47 @@ def test_positive_allow_reregistration_when_dmi_uuid_changed(
     ).create()
     result = rhel_contenthost.execute(command)
     assert result.status == 0
+
+
+def test_positive_update_packages_post_registration(
+    module_target_sat,
+    module_entitlement_manifest_org,
+    module_location,
+    module_ak,
+    rhel8_contenthost,
+):
+    """Test package update on host post registration
+
+    :id: 3d0a3252-ab81-4acf-bca6-253b746f26bb
+
+    :expectedresults: Package update is successful on host post registration.
+
+    :CaseLevel: Integration
+    """
+    org = module_entitlement_manifest_org
+
+    rh_repo_id = module_target_sat.api_factory.enable_rhrepo_and_fetchid(
+        basearch=constants.DEFAULT_ARCHITECTURE,
+        org_id=module_entitlement_manifest_org.id,
+        product=constants.PRDS['rhel8'],
+        repo=constants.REPOS['rhst8']['name'],
+        reposet=constants.REPOSET['rhst8'],
+        releasever=None,
+    )
+    rh_repo = module_target_sat.api.Repository(id=rh_repo_id).read()
+    rh_repo.sync()
+
+    result = rhel8_contenthost.register(
+        org,
+        module_location,
+        module_ak.name,
+        module_target_sat,
+        update_packages=True,
+    )
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+
+    package = constants.FAKE_7_CUSTOM_PACKAGE
+    repo_url = settings.repos.yum_3['url']
+    rhel8_contenthost.create_custom_repos(fake_yum=repo_url)
+    result = rhel8_contenthost.execute(f"yum install -y {package}")
```

Comment:
Why doing this post-registration explicitely when all packages are updated?

Reply:
This makes me think about if the update_packages is marked as true will the package update be successful? if yes how and what's its significance?

Reply:
@Gauravtalreja1 The package update will be different from installing a package on client post registration.
@adarshdubey-star with update_packages set to true, it updates any existing package available on client. The second scenario would be to install new package on client system.

Reply:
@shweta83 apologies for confusing comment, I mean while registering we could enable this single repo on host and install packages using available options, instead of explicitely doing this after registration

Reply:
Ohh ok. We need to test package installation after registration.